### DISCUSSION TITLE
Allow multiple reply threads to any message

### DISCUSSION THREAD

**Discussion by gigawhitlocks** [mattermost/community#61] [Moonshot Ideas]
Today, any given channel-level post may have 1 child thread, and thread posts may have 0 thread children.

Clearly, this is easier to design and build, from both a UI/UX and backend perspective, so it makes sense as to why this is the dominant threading paradigm present not only in Mattermost but in other pieces of communication software as well.

But this model doesn't accurately represent the reality of asynchronous many-to-many textual discussion. The truth is that at any point, a user might want to 'reply' to a post, starting a new thread, and that post may be in a thread or in the main channel and that distinction is not important to the user or to the conversation in the abstract.

It would be interesting if a new thread could be spawned as a reply to any post. For simplicity, one can imagine that the new thread would still be at the channel level, but with a link back to the parent post, either in the main channel or in a different thread.

**Comment by esethna** [comment]
Makes sense, definitely heard of this in the past. Does the idea to pop-out into a new thread solve this use case in your mind? https://github.com/mattermost/community/discussions/21#discussioncomment-905785
